subprojects/docs/src/docs/userguide/directory_layout.adoc

```diff
@@ -0,0 +1,104 @@
+// Copyright 2017 the original author or authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+[[directory_layout]]
+= Directory Layout
+
+[.lead]
+Gradle uses two main directories to perform and manage its work: the <<#directory_layout:gradle_user_home_directory>> and the <<#directory_layout:project_root_directory>>.
+
+
+[[directory_layout:gradle_user_home_directory]]
```

Comment:
Links from other chapters need to include the name of the chapter file anyway, so the `directory_layout:` prefix is kind of redundant. It does help keep the anchor unique for the single-page HTML and PDF documents, but perhaps `dir:` would be sufficient?